compile error

Reading the automatic build log of @jperon for Ubuntu, I've seen this error:

```
make[2]: Entering directory '/home/eroux/gregorio/gregorio/src'
gcc -DHAVE_CONFIG_H -DBRANCH_VERSION='"develop-593cdca-2397"' -I.  -I../src -I../src/gabc  -I../src/dump  -I../src/gregoriotex -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2   -g -O2 -std=gnu89 -pedantic-errors -fstack-protector-strong -fPIE -Wformat -Werror=format-security -Wstrict-prototypes -Wall -Wextra -MT gregorio-gregorio-utils.o -MD -MP -MF .deps/gregorio-gregorio-utils.Tpo -c -o gregorio-gregorio-utils.o `test -f 'gregorio-utils.c' || echo './'`gregorio-utils.c
gregorio-utils.c: In function ‘main’:
gregorio-utils.c:26:27: error: passing argument 1 of ‘gregorio_set_file_name’ discards ‘const’ qualifier from pointer target type
 #define gregorio_basename xbasename
                           ^
gregorio-utils.c:563:32: note: in expansion of macro ‘gregorio_basename’
         gregorio_set_file_name(gregorio_basename(input_file_name));
                                ^
In file included from gregorio-utils.c:37:0:
messages.h:52:6: note: expected ‘char *’ but argument is of type ‘const_string’
 void gregorio_set_file_name(char *new_name);
```

which I can reproduce easily with the `--with-kpathsea` option to `configure`. Do you know where it comes from?

--- a/src/messages.c
+++ b/src/messages.c
@@ -25,7 +25,7 @@
 #include "messages.h"
 
 static FILE *error_out;
-static char *file_name = NULL;
+static const char *file_name = NULL;
 static gregorio_verbosity verbosity_mode = 0;
 static bool debug_messages = false;
 static int return_value = 0;
@@ -40,7 +40,7 @@ void gregorio_set_error_out(FILE *const f)
     error_out = f;
 }
 
-void gregorio_set_file_name(char *const new_name)
+void gregorio_set_file_name(const char *const new_name)
 {
     file_name = new_name;
 }

--- a/src/messages.h
+++ b/src/messages.h
@@ -49,7 +49,7 @@ void gregorio_messagef(const char *function_name,
         int line_number, const char *format, ...)
         __attribute__ ((__format__ (__printf__, 4, 5)));
 void gregorio_set_verbosity_mode(gregorio_verbosity verbosity);
-void gregorio_set_file_name(char *new_name);
+void gregorio_set_file_name(const char *new_name);
 void gregorio_set_error_out(FILE *f);
 int gregorio_get_return_value(void);
 

